tsdb/head.go

```diff
@@ -763,8 +763,15 @@ func NewRangeHead(head *Head, mint, maxt int64) *RangeHead {
 	}
 }
 
-func (h *RangeHead) Index() (IndexReader, error) {
-	return h.head.indexRange(h.mint, h.maxt), nil
+func (h *RangeHead) Index(mint, maxt int64) (IndexReader, error) {
+	// rangeHead guarantees that the series returned are within its range.
+	if mint < h.mint {
```

Comment:
This logic seems the wrong way around.

Reply:
That is cropping the Index mint and maxt to the RangeHead's mint and maxt. It seems right to me. This seems to be the purpose of the RangeHead.

Reply:
Ah yes